SofaKernel/modules/Sofa.Geometry/src/sofa/geometry/Triangle.h

```diff
@@ -70,6 +70,26 @@ struct Triangle
             return static_cast<T>(0.25) * std::sqrt((a + b + c) * (-a + b + c) * (a - b + c) * (a + b - c));
         }
     }
+
+
+    template<typename Node,
+        typename T = std::decay_t<decltype(*std::begin(std::declval<Node>()))>,
+        typename = std::enable_if_t<std::is_scalar_v<T>>
+    >
+        [[nodiscard]]
+    static constexpr auto normal(const Node& n0, const Node& n1, const Node& n2)
+    {
+        if constexpr (std::is_same_v < Node, sofa::type::Vec<3, T> >)
+        {
+            sofa::type::Vec<3, T> normal_t = (n1 - n0).cross(n2 - n0);
+            return normal_t;
+        }
+        else /*if constexpr (std::is_same_v< Node, sofa::type::Vec<2, T> >)*/
+        {
+            // shoelace formula
+            return sofa::type::Vec<3, T>();
```

Comment:
do you need this case?

Reply:
Agreed, I dont think you should be allowed to call normal if you are not in 3D (or more ?)
you could add at the beginning:
```
        constexpr Node n{};
        static_assert(std::distance(std::begin(n), std::end(n)) == 3, "volume can only be computed in 3 dimensions.");
```
Actually, you will have a valid normal only if Node=Vec3 so either you directly use Vec in the signature of the function, or in the else of the constexpr, you do a generic computation to get a normal. This could be bothersome :p